solidity/contracts/deposit/DepositUtils.sol

```diff
@@ -448,8 +448,14 @@ library DepositUtils {
     /// @param _redeemer    The assumed owner of the deposit's TDT.
     /// @return             The amount in TBTC needed to redeem the deposit.
     function getOwnerRedemptionTbtcRequirement(DepositUtils.Deposit storage _d, address _redeemer) internal view returns(uint256) {
-        uint256 frtAdjust = frtAdjust(_d, _redeemer);
-        uint256 signerFeeAdjust = signerFeeAdjust(_d, _redeemer);
+        bool inCourtesy = _d.inCourtesyCall();
+        bool atTerm = remainingTerm(_d) == 0;
+        bool redeemerPaysNoFee = inCourtesy || atTerm;
+        bool frtOwnerRedemption = feeRebateTokenHolder(_d) == _redeemer;
+        bool frtExists = feeRebateTokenHolder(_d) != address(0);
+
+        uint256 frtAdjust = computeFrtAdjustment(_d, redeemerPaysNoFee, frtOwnerRedemption, frtExists);
+        uint256 signerFeeAdjust = computeSignerFeeAdjustment(_d, redeemerPaysNoFee, frtExists);
```

Comment:
Am I reading this wrong that this is identical to `getRedemptionTbtcRequirement` except with `tdtOwnerRedemption` always `true`? If so, I think we can stack these?

```solidity
function getOwnerRedemptionTbtcRequirement(DepositUtils.Deposit storage _d, address _redeemer) internal pure returns (uint256) {
    // The TDT owner only ever owes fee adjustments.
    return computeRedemptionFeeAdjustments(_d, _redeemer);
}

function getRedemptionTbtcRequirement(DepositUtils.Deposit storage _d, address _redeemer) internal view returns (uint256) {
    bool redeemerHoldsTdt = depositOwner(_d) == _redeemer;
    bool inCourtesy = _d.inCourtesyCall();
    bool atTerm = remainingTerm(_d) == 0;

    require(redeemerHoldsTdt || inCourtesy || atTerm, "Only TDT holder can redeem unless deposit is at-term or courtesy called");

    uint256 mainCharge = computeBaseRedemptionCharge(_d, redeemerHoldsTdt);    
    return mainCharge.add(computeRedemptionFeeAdjustments(_d, _redeemer));
}

function computeRedemptionFeeAdjustments(DepositUtils.Deposit storage _d, address _redeemer) internal pure returns (uint256) {
    bool inCourtesy = _d.inCourtesyCall();
    bool atTerm = remainingTerm(_d) == 0;
    bool redeemerOwesNoFee = inCourtesy || atTerm;
    bool redeemerHoldsFrt = feeRebateTokenHolder(_d) == _redeemer;
    bool frtExists = feeRebateTokenHolder(_d) != address(0);

    uint256 frtAdjustment = computeFrtAdjustment(_d, redeemerOwesNoFee, redeemerHoldsFrt, frtExists)
    uint256 signerFeeAdjustment = computeSignerFeeAdjustment(_d, redeemerOwesNoFee, frtExists)

    return frtAdjustment.add(signerFeeAdjustment)
}
```

Reply:
In fact, I would go so far as to make even `getRedemptionTbtcRequirement` be `pure`, and move the redeemerHoldsTdt || inCourtesy || atTerm require all the way up to `_requestRedemption`, honestly, but that may be too much right now.

Reply:
I'm not sure if grouping the fee adjustment functions is good. Won't it just make things more expensive? also not sure if it helps in terms of clarity

Reply:
Having two paths that do the exact same thing is definitely counter to clarity heh. We want our code to reflect the path similarity, and the split between getOwner and getRedemptionTbtcRequirement is artificial in the first place in service of the vending machine. It's considerably more difficult without this change to look at the code and understand just how similar these paths are, IMO.

Reply:
> It's considerably more difficult without this change to look at the code and understand just how similar these paths are, IMO.

you're right here

Reply:
re-calculating those bools within the same function chain bugs me a little

Reply:
I agree 😬 Just didn’t see an immediately obvious cleanup. I think it’s reflective of the broader problem of *needing* these two functions, tbh. The reason we need them is that `require`, which we don’t want to trigger in the owner function. The owner function is a simple calculation, the other one is imposing restrictions. This is where pulling the `require` further up would allow this code to be focused purely on calculation, but it also has some knock on effects that we may not want to play all the way out. If you want to give it a shot though, you’re welcome to :)